### PR TITLE
Add script to generate diffs for the hbs templates

### PR DESCRIPTION
**Pull request by steplov**
Simple script which generates html reports with a diffs in handlebar templates. It should save a lot of time for the developers

### CODE REVIEW

File: build-tools/script/diff2html.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  {
+    type: 'input',
+    name: 'commitHash',
+    message: 'Please enter a commit hash',
+    when: answers => answers.type === 'hash',
+    validate: val => val.length > 0,
+  },
+];
+
+const createDir = () => {
```
Comment (ThaNarie) [inline build-tools/script/diff2html.js]: Could we use [fs-extra](https://github.com/jprichardson/node-fs-extra) for this? Has methods to easily create files/folders without having to check.

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "dev": "webpack-dev-server --config build-tools/config/webpack/webpack.config.js",
     "dev:code": "webpack --config build-tools/config/webpack/webpack.config.code.dev.js --watch",
     "build": "cross-env NODE_ENV=production node ./build-tools/script/build",
+    "postbuild": "cross-env node ./build-tools/script/diff2html",
```
Comment (ThaNarie) [inline package.json]: I think this script should not run automatically when running a build (build is run very often, but a delivery only every x days/weeks), so should be invoked manually.

Comment (steplov) [reply package.json]: Ok. I think i will call it "build:diff" if you don't mind

Comment (ThaNarie) [reply package.json]: Sounds good!

File: build-tools/script/diff2html.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    for (let i = 1; i <= path.length; i++) {
+      const segment = path.slice(0, i).join('/');
+
+      !fs.existsSync(segment) ? fs.mkdirSync(segment) : null;
+    }
+  }
+};
+
+inquirer.prompt(questions).then(answers => {
+  const diff = answers.type === 'hash' ? answers.commitHash : 'origin/master';
```
Comment (ThaNarie) [inline build-tools/script/diff2html.js]: Could we make `diff` a range (`from..to`), so we have more control over where to compare it to?

This also means we ask two questions (where the to defaults to `HEAD`).

**Comment by ThaNarie** [comment]
As talked about offline, it would be very nice if we could store each diff-creation in a meta file (with commit hash, commit message and a 'added' timestamp, and store in the repo), and use that as selection option to compare the current version with.

Unused builds can be easily removed from the file by developers if it was never used.

File: build-tools/script/diff2html.js [hunk 4]
```diff
@@ -9,33 +9,56 @@ const EXT = '*.hbs';
...
-      {
-        name: 'Commit hash',
-        value: 'hash',
-      },
       {
         name: 'Master branch',
         value: 'master',
       },
+      {
+        name: 'Commit hash or branch name',
```
Comment (ThaNarie) [inline build-tools/script/diff2html.js]: or tag :)

File: build-tools/script/diff2html.js [hunk 5]
```diff
@@ -9,33 +9,56 @@ const EXT = '*.hbs';
...
   },
 ];
 
 inquirer.prompt(questions).then(answers => {
-  const diff = answers.type === 'hash' ? answers.commitHash : 'origin/master';
+  const from = answers.from === 'value' ? answers.fromValue : 'origin/master';
+  const to = answers.from === 'value' ? answers.toValue : '';
 
-  const command = `diff2html -s side -f html -F ${OUTPUT_DIR}/${REPORT_NAME} -- -M ${diff} -- ${EXT}`;
+  const command = `diff2html -s side -f html -F ${OUTPUT_DIR}/${REPORT_NAME} -- -M ${from} ${to} -- ${EXT}`;
```
Comment (ThaNarie) [inline build-tools/script/diff2html.js]: Is this correct diff syntax? Shouldn't it be `${from}..${to}` ?

Comment (steplov) [reply build-tools/script/diff2html.js]: Yes. `${from}..${to}` is synonymous to `${from} ${to}` https://git-scm.com/docs/git-diff#git-diff-emgitdiffem--optionsltcommitgtltcommitgt--ltpathgt82308203

File: build-tools/script/diff2html.js (hunk 4 above)
Comment (steplov) [reply build-tools/script/diff2html.js]: We can simplify it and keep only <commit> like in official documentation.

**Pull request merged**